src/producer/mod.rs

```diff
@@ -122,20 +122,21 @@ impl<S: Read + Write> Producer<S> {
         &mut self,
         jobs: J,
     ) -> Result<(usize, Option<HashMap<String, String>>), Error>
-      where J: IntoIterator<Item = Job>,
-            J::IntoIter: ExactSizeIterator
+    where
+        J: IntoIterator<Item = Job>,
+        J::IntoIter: ExactSizeIterator,
     {
         let jobs = jobs.into_iter();
-        let njobs = jobs.len();
+        let jobs_count = jobs.len();
         let errors: HashMap<String, String> = self
             .c
-            .issue(&PushBulk::from(jobs))?
+            .issue(&PushBulk::from(jobs.collect::<Vec<_>>()))?
```

Comment:
I'm hoping we can maybe avoid the `collect` here too. The trick is going to be to also make `PushJobs` generic (see comment there).